script.js

```diff
@@ -9,123 +11,71 @@ $(document).ready(function () {
-            if (ingredients6 === null) {
-                ingredients6 = '';
-            }
-            // diplays drink name in classic-drink-title area
-            $("#classic-drink-title").text(`${name}`);
-            // creates glass name on new line
-            let glassName = $("<p>").text(`Glass: ${glass}`);
-            // creates instructions on new line
-            let drinkInstructions = $("<p>").text(`Instructions: ${instructions}`);
-            // creates ingredients on new line
-            let ingredients = $("<p>").text(`Ingredients: ${ingredients1} ${ingredients2} ${ingredients3} ${ingredients4} ${ingredients5} ${ingredients6}`);
-            // apppends all info to drinksDiv
-            drinksDiv.append(image, glassName, ingredients, drinkInstructions);
+            //           drink to display  |       title element     |     description element
+            displayDrink(response.drinks[0], $("#classic-drink-title"), $(".classic-drinks"));
         });
     }
-    // on click event for classic drink btns
-    $(document).on("click", ".drink", classicDrinkButtons);
 
+    /*
+        Both the random drink and the classic drinks have a lot of shared behavior.
+        this 'displayDrink' method encapsulates that. From what was previously here,
+        all we need are the 2 DOM elements for the drink title, and the drink description
+        area where the picture, ingredients, glass, and instructions display.
+    */
+    function displayDrink(drink, titleElement, descriptionElement) {
+        let ingredientsList = getIngredients(drink);
+        
+        const ingredientStr = ingredientsList
+            // .map(x => x.measure == null ? '' x.measure + x.ingredient) // If we want the measurement
+            .map(x => x.ingredient)
+            .join(" ");
+        
+        let randomName        = $("<p>").text(`${drink.strDrink}`);
+        let ingredients       = $("<p>").text(`Ingredients: ${ingredientStr}`);
+        let glassName         = $("<p>").text(`Glass: ${drink.strGlass}`);
+        let drinkInstructions = $("<p>").text(`Instructions: ${drink.strInstructions}`);
+        var image             = $("<img>").attr("src", drink.strDrinkThumb);
+
+        titleElement      .append(randomName);
+        descriptionElement.append(image, glassName, ingredients, drinkInstructions);
+    }
+
+    function getIngredients(drink) {
+        let ingredientsList = [];
+        var i;
+        for(i = 1; i <= 15; i++) {
+            // The API doesn't return a json list, so we have to do this nonsense
+            // to keep the ingredients in line with the measurements
+            var recipe = {
+                ingredient: drink["strIngredient" + i],
+                measure:    drink["strMeasure"    + i]
+            }
+
+            if(recipe.ingredient != null)
+                ingredientsList.push(recipe);
+        }
+        return ingredientsList;
+    }
```

Comment:
The json returned from cocktailDB is dumb... That's an issue with the API though, not on you.
```
{
   "drinks":[
      {
         "idDrink":"17833",
         "strDrink":"A. J.",
          ...
         "strIngredient1":"Applejack",
         "strIngredient2":"Grapefruit juice",
         "strIngredient3":null,
         "strIngredient4":null,
          ...
         "strMeasure1":"1 1/2 oz ",
         "strMeasure2":"1 oz ",
         "strMeasure3":null,
         "strMeasure4":null,
         ...
      }
   ]
}
```

Really should look something more like this so that an ingredient is clearly grouped with its measurement, and we don't have to sift through a bunch of `null` ingredients for those that have fewer than 15 ingredients.

```
{
   "drinks":[
      {
         "idDrink":"17833",
         "strDrink":"A. J.",
          ...
         "ingredients": [
                 {
                        "ingredient": "Applejack",
                        "measurement": "1 1/2 oz "
                 },
                 {
                        "ingredient": "Grapefruit juice",
                        "measurement": "1 oz "
                 },
         ]
         ...
      }
   ]
}
```